### PR TITLE
fix(libsinsp): display invalid pids as <NA>

### PR DESCRIPTION
**Pull request by LucaGuerra**
**What type of PR is this?**

> Uncomment one (or more) `/kind <>` lines:

/kind bug

**Any specific area of the project related to this PR?**

> Uncomment one (or more) `/area <>` lines:

/area libsinsp

**Does this PR require a change in the driver versions?**

> /version driver-API-version-major

> /version driver-API-version-minor

> /version driver-API-version-patch

> /version driver-SCHEMA-version-major

> /version driver-SCHEMA-version-minor

> /version driver-SCHEMA-version-patch

**What this PR does / why we need it**:

Sometimes PIDs do not necessarily indicate host PIDs. This is true if we're monitoring sandboxes which do not map to a real kernel, such as gVisor. In order to maintain internal state consinstency sinsp will hold a PID that is larger than an actual one for those processes which acts as a unique identifier. However, those are not meant to be consumed by users.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
fix(libsinsp): display invalid PIDs as <NA>
```

### CODE REVIEW

**Review by leogr — APPROVED** [review]

File: userspace/libsinsp/filterchecks.cpp [hunk 1]
```diff
@@ -2142,6 +2142,14 @@ uint8_t* sinsp_filter_check_thread::extract_thread_cpu(sinsp_evt *evt, OUT uint3
 	return NULL;
 }
 
+// Some syscall sources, such as the gVisor integration, cannot match events to host PIDs and TIDs.
+// The event will retain the PID field which is consistent with the rest of sinsp logic, but it won't represent
+// a real PID and so it should not be displayed to the user.
+inline bool should_extract_xid(int64_t xid)
+{
+	return xid >= -1 && xid <= UINT32_MAX;
```
Comment (Andreagit97) [inline userspace/libsinsp/filterchecks.cpp:2150]: Why also - 1 here? Is this valid in our logic?

**Review by Andreagit97 — COMMENTED** [review]
Just a question in the comment

File: userspace/libsinsp/filterchecks.cpp (hunk 1 above)
Comment (LucaGuerra) [reply userspace/libsinsp/filterchecks.cpp:2150]: There may be cases in which we do return -1 if sinsp cannot determine the correct thread ID but I'm not sure. In any case, I don't want to introduce breaking changes :)

**Review by Andreagit97 — APPROVED** [review]
/approve

**Pull request merged**